### PR TITLE
Configure GH Actions CI to run tests and build on Windows machines

### PR DESCRIPTION
**Pull request by RishiKumarRay**
#5

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -48,6 +50,10 @@ jobs:
...
           # refresh cache every month to avoid unlimited growth
           key: maven-repo-${{ runner.os }}-${{ steps.get-date.outputs.date }}
+          
+      - name: Prepare git
+        run: git config --global core.autocrlf false
+      - uses: actions/checkout@v2  
 
       - name: Build with Maven
-        run: mvn -B formatter:validate verify --file pom.xml
+        run: mvn -B formatter:validate verify -D"format.skip"=true --file pom.xml
```
Comment (gastaldi) [inline .github/workflows/build.yml:59]: You can revert this change

**Comment by RishiKumarRay** [comment]
@gastaldi done

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -48,6 +50,10 @@ jobs:
...
           # refresh cache every month to avoid unlimited growth
           key: maven-repo-${{ runner.os }}-${{ steps.get-date.outputs.date }}
+          
+      - name: Prepare git
+        run: git config --global core.autocrlf false
+      - uses: actions/checkout@v2  
 
       - name: Build with Maven
-        run: mvn -B formatter:validate verify --file pom.xml
+        run:  mvn -B formatter:validate verify --file pom.xml
```
Comment (gastaldi) [inline .github/workflows/build.yml:59]: This extra space is also unnecessary ;)

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -48,6 +50,10 @@ jobs:
           path: ~/.m2/repository
           # refresh cache every month to avoid unlimited growth
           key: maven-repo-${{ runner.os }}-${{ steps.get-date.outputs.date }}
+          
+      - name: Prepare git
```
Comment (gastaldi) [inline .github/workflows/build.yml:54]: This needs to happen before Line 31

**Comment by RishiKumarRay** [comment]
@gastaldi updated that too, maybe that is why it was not passing ?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -24,12 +24,16 @@ on:
...
   build:
-
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.os }}
+    strategy:
+     matrix:
+        os: [ubuntu-latest, windows-latest]
     steps:
+      - uses: actions/checkout@v2
```
Comment (gastaldi) [inline .github/workflows/build.yml:32]: Remove this line

**Comment by ricardozanini** [comment]
@RishiKumarRay, thanks for the PR! Like I said before, it won't pass until I fix #7 since we have a bug on the Windows machine. That's good. As soon as I set it, I can rerun this action to verify if it was, indeed, fixed.

**Comment by RishiKumarRay** [comment]
@gastaldi sorry for these many mistakes , is it fine now?

**Comment by gastaldi** [comment]
@RishiKumarRay although there are still some changes in the lines that are not related to this enhancement, it looks okay 👍🏻

**Review by gastaldi — APPROVED** [review]
Awesome job! This now reproduces the build error in Windows: https://github.com/quarkiverse/quarkus-openapi-generator/runs/5219495336?check_suite_focus=true

**Comment by ricardozanini** [comment]
Thanks guys! I'll set up a VM to fix the error.

**Comment by ricardozanini** [comment]
@RishiKumarRay, PR #12 should solve this issue. As soon as we merge it, we can give it a try to your changes.

**Comment by ricardozanini** [comment]
"works on my machine"...

**Comment by ricardozanini** [comment]
I forgot to update this branch 🤦

**Comment by ricardozanini** [comment]
@all-contributors please add @gastaldi for infrastructure and code
@all-contributors please add @RishiKumarRay for infrastructure

**Comment by ricardozanini** [comment]
https://github.com/all-contributors please add @gastaldi  for code and infra

**Comment by ricardozanini** [comment]
@all-contributors please add @RishiKumarRay for infra

**Comment by ricardozanini** [comment]
@all-contributors please add @gastaldi for code and infra

**Comment by gastaldi** [comment]
Yeah, the bot isn't that smart. 
Thanks for the mention, though :)
